Sources/Views/ALKChatBaseCell.swift

```diff
@@ -91,7 +91,47 @@ open class ALKChatBaseCell<T>: ALKBaseCell<T>, Localizable {
             menuController.setMenuVisible(true, animated: true)
         }
     }
+    
+    // To show Menu Controller if user long presses the Link
+    func showMenuControllerForLink(_ gestureView : UIView) {
+        NotificationCenter.default.addObserver(self, selector: #selector(menuWillShow(_:)), name: UIMenuController.willShowMenuNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(menuWillHide(_:)), name: UIMenuController.willHideMenuNotification, object: nil)
+
+        if let chatBar = chatBar, chatBar.textView.isFirstResponder {
+            chatBar.textView.overrideNextResponder = contentView
+        } else {
+            _ = canBecomeFirstResponder
+        }
+
+        guard let superView = gestureView.superview else {return}
+        
+        let menuController = UIMenuController.shared
+
+        guard !menuController.isMenuVisible, gestureView.canBecomeFirstResponder else {
+            return
+        }
 
+        gestureView.becomeFirstResponder()
+
+        var menus: [UIMenuItem] = []
+
+        if let copyMenu = getCopyMenuItem(copyItem: self) {
+            menus.append(copyMenu)
+        }
+
+        if let replyMenu = getReplyMenuItem(replyItem: self) {
+            menus.append(replyMenu)
+        }
+
+        if showReport, let reportMessageMenu = getReportMessageItem(reportMessageItem: self) {
+            menus.append(reportMessageMenu)
+        }
+
+        menuController.menuItems = menus
+        menuController.setTargetRect(gestureView.frame, in: superView)
+        menuController.setMenuVisible(true, animated: true)
```

Comment:
If the goal is to enable long tap messages for link messages, then can we do these changes in `ALKFriendLinkPreviewCell` vs applying to all cells?

Reply:
moved codes to ALKLinkPreviewCell